src/printer.js

```diff
@@ -23,18 +23,8 @@ const warnDeprecation = once(() => {
   return true;
 });
```

Comment:
If this PR is explicitly dropping support for v2, I wonder if this should be kept, specifying `3.0.0` instead?

Reply:
Same question for:

- `src/common/backward-compatibility.js`
- `src/slangPrinter.ts`
- `tests/config/get-plugins.js`

Reply:
prettier 2.2.X would still load the plugin and fail miserably when using it, so we needed the check for `>=2.3.0` to show the reasons why it was failing.
Now prettier 2.X.X won't even load the plugin so the code will never reach this code.
there is a check in `peerDependencies` in `package.json` and I'll add this info on the readme.